react/features/analytics/AnalyticsEvents.js

```diff
@@ -1,663 +1,474 @@
+ * the jitsi-meet API.
  *
- * @type {String}
+ * @param {Object} action - The action which was requested through the
+ * jitsi-meet API.
+ * @param {Object} attributes - Attributes to attach to the event.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_SCREEN_TOGGLED = 'shortcut.screen.toggled';
+export const createApiEvent = function(action, attributes = {}) {
+    return {
+        action,
+        attributes,
+        source: 'jitsi-meet-api'
+    };
+};
 
 /**
- * Toggled the display of the speaker stats dialog.
+ * Creates an event which indicates that the audio-only mode has been turned
+ * off.
  *
- * @type {String}
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_SPEAKER_STATS_CLICKED = 'shortcut.speakerStats.clicked';
+export const createAudioOnlyDisableEvent = function() {
+    return {
+        name: 'audio.only.disabled'
+    };
+};
 
 /**
- * Started pressing the key that undoes audio mute while the key is pressed.
+ * Creates an event which indicates that a device was changed.
  *
- * @type {String}
+ * @param {string} mediaType - The media type of the device ('audio' or
+ * 'video').
+ * @param {string} deviceType - The type of the device ('input' or 'output').
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_TALK_CLICKED = 'shortcut.talk.clicked';
+export const createDeviceChangedEvent = function(mediaType, deviceType) {
+    return {
+        name: 'device.changed',
+        attributes: {
+            deviceType,
+            mediaType
+        }
+    };
+};
 
 /**
- * Released the key used to talk while audio muted, returning to the audio muted
- * state.
+ * Creates an event which specifies that the feedback dialog has been opened.
  *
- * @type {String}
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_TALK_RELEASED = 'shortcut.talk.released';
+export const createFeedbackOpenEvent = function() {
+    return {
+        name: 'feedback.opened'
+    };
+};
 
 /**
- * Toggling video mute state using a keyboard shortcut.
+ * Creates a "filmstrip toggled" event.
  *
- * @type {String}
+ * @param {Object} attributes - Attributes to attach to the event.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_VIDEO_MUTE_TOGGLED = 'shortcut.videomute.toggled';
+export const createFilmstripToggledEvent = function(attributes) {
+    return {
+        attributes,
+        name: 'filmstrip.toggled'
+    };
+};
 
 /**
- * The config specifies the local participant should start with audio only mode
- * enabled or disabled.
+ * Creates an event which indicates that the invite dialog was closed. This is
+ * not a UI, since it is not necessarily the result of a user interaction.
```

Comment:
I think it'd be better to say not a TYPE_UI instead of UI, because the term "UI" already has different meanings...and as you learned doesn't even mean what we thought it meant.

Reply:
Good point